### PR TITLE
Add Q.noConflict function

### PR DESCRIPTION
**Pull request by kahnjw**
Fixes https://github.com/kriskowal/q/issues/677

This commit adds a function that is basically a noop. It simply alerts the user that they should not be using `Q.noConflict` when they are not using Q as a global.

If Q finds itself sourced as a `<script>` it will store the original value of `Q` and overwites `Q.noConflict` function with an actual implementation. `Q.noConflict` returns a refernce to `Q`.

@kriskowal didn't know if adding the no-op function to `Q` if we're in a non-browser environment was preferable, but it seemed right for consistency, and testability. Let me know what you think.

### CODE REVIEW

File: q.js [hunk 1]
```diff
@@ -2022,6 +2032,10 @@ Promise.prototype.nodeify = function (nodeback) {
     }
 };
 
+Q.noConflict = function() {
+    console.log('Q.noConflict only works when Q is used as a global');
```
Comment (kriskowal) [inline q.js]: This should simply throw an error.

File: spec/q-spec.js [hunk 2]
```diff
@@ -2491,6 +2491,27 @@ describe("node support", function () {
...
+
+    afterEach(function() {
+        Q = _Q;
+    });
+
+    it("sets the global Q object to its original value", function() {
+        Q.noConflict();
+
+        // In this context the original value of Q is undefined.
+        if(typeof window !== 'undefined') {
```
Comment (kriskowal) [inline spec/q-spec.js]: Attend to the prevailing style with regard to spaces. I believe `npm run lint` will inform, but if it does not, take care to place spaces between keywords and parens to distinguish such cases from function calls.

**Comment by kriskowal** [comment]
This seems reasonable and I would land this unless @domenic has strong opinions to the contrary. Bother me if I take no action between now and mid May. I do my open source gardening on weekends.

**Comment by kahnjw** [comment]
Will do. I probably won't have time to update until tomorrow anyway.

**Comment by kahnjw** [comment]
>mid May

Whops, misread this as "mid day." Got it.

I've updated the pull.

**Pull request merged**